mm2src/coins/eth/eth_swap_v2/mod.rs

```diff
@@ -108,6 +120,112 @@ impl EthCoin {
+            while next_from_block <= current_block {
+                let to_block = std::cmp::min(next_from_block + self.logs_block_range - 1, current_block);
+
+                // Fetch events for the current block range
+                let events = match self
+                    .events_from_block(
+                        params.swap_contract_address,
+                        params.event_name,
+                        next_from_block,
+                        Some(to_block),
+                        params.abi_contract,
+                    )
+                    .await
+                {
+                    Ok(events) => events,
+                    Err(e) => {
+                        error!(
+                            "Error getting {} events from {} to {} block: {}",
+                            params.event_name, next_from_block, to_block, e
+                        );
+                        Timer::sleep(params.check_every).await;
+                        // Move to next window if there was an error
+                        next_from_block += self.logs_block_range;
+                        continue;
+                    },
+                };
+
+                // Check if any event matches the SWAP ID
+                if let Some(found_event) = events
+                    .into_iter()
+                    .find(|event| event.data.0.len() >= 32 && &event.data.0[..32] == params.swap_id)
+                {
+                    if let Some(hash) = found_event.transaction_hash {
+                        return Ok(hash);
+                    }
+                }
+
+                next_from_block += self.logs_block_range;
+            }
+
+            Timer::sleep(params.check_every).await;
+        }
+    }
+
+    /// Waits until the specified transaction is found by its hash or the given timeout is reached
+    pub(crate) async fn wait_for_spend_transaction(
+        &self,
+        tx_hash: H256,
+        wait_until: u64,
+        check_every: f64,
+    ) -> MmResult<SignedEthTx, FindPaymentSpendError> {
+        loop {
+            let now = now_sec();
+            if now > wait_until {
+                return MmError::err(FindPaymentSpendError::Timeout { wait_until, now });
+            }
+
+            match self.transaction(TransactionId::Hash(tx_hash)).await {
+                Ok(Some(t)) => {
+                    let transaction = signed_tx_from_web3_tx(t).map_err(FindPaymentSpendError::Internal)?;
+                    return Ok(transaction);
+                },
+                Ok(None) => info!("Transaction {} not found yet", tx_hash),
+                Err(e) => error!("Get transaction {} error: {}", tx_hash, e),
+            };
+
+            Timer::sleep(check_every).await;
+        }
+    }
+}
+
+pub(crate) struct SpendTxSearchParams<'a> {
+    pub(crate) swap_contract_address: Address,
+    pub(crate) event_name: &'a str,
+    pub(crate) abi_contract: &'a Contract,
+    pub(crate) swap_id: &'a [u8],
```

Comment:
can we add size type to this if possible

Reply:
yes both legacy and tpu eth swaps use 32 byte swap id
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/b04f217df034e491b8f851a1480a3a85cac05999